app/src/ui/merge-conflicts/merge-conflicts-dialog.tsx

```diff
@@ -238,17 +238,12 @@ export class MergeConflictsDialog extends React.Component<
   private renderUnmergedFile(
     file: WorkingDirectoryFileChange
   ): JSX.Element | null {
-    switch (file.status) {
-      case AppFileStatus.Resolved:
+    const { status } = file
+    switch (status.kind) {
+      case AppFileStatusKind.Resolved:
         return this.renderResolvedFile(file.path)
-      case AppFileStatus.Conflicted:
-        if (file.conflictStatus === null) {
-          throw new Error(
```

Comment:
This was the main benefit of the PR - being able to verify at compile time that we're in a known good state

Reply:
💯